lib/grandpa/message_handler.go

```diff
@@ -661,30 +662,30 @@ func (s *Service) VerifyBlockJustification(hash common.Hash, justification []byt
 	}
 
 	if count+len(equivocatoryVoters) < threshold {
-		return ErrMinVotesNotMet
+		return nil, ErrMinVotesNotMet
 	}
 
 	err = verifyBlockHashAgainstBlockNumber(s.blockState, fj.Commit.Hash, uint(fj.Commit.Number))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	for _, preCommit := range fj.Commit.Precommits {
 		err := verifyBlockHashAgainstBlockNumber(s.blockState, preCommit.Vote.Hash, uint(preCommit.Vote.Number))
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	err = s.blockState.SetFinalisedHash(hash, fj.Round, setID)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	logger.Debugf(
 		"set finalised block with hash %s, round %d and set id %d",
 		hash, fj.Round, setID)
-	return nil
+	return scale.Marshal(fj)
```

Comment:
A little confused, just for my own understanding where are the extra bytes trimmed here? Is it just by unmarshalling then marshalling?

Reply:
Yes, the extra bytes are removed when the un-marshaled Justification is marshaled.